File: BubbleSort/bubbleSort.c
#include <stdio.h>


void bubbleSort(int* list, int size) {
	for (int i = 0; i < size; i++) {
		for (int j = 0; j < size - 1; j++) {
			if (list[j] > list[j + 1]) {
				int temp = list[j];
				list[j] = list[j + 1];
				list[j + 1] = temp;
			}
		}
	}
}

void print(int* list, int size) {
	for (int i = 0; i < size; i++) {
		printf("%d ", list[i]);
	}
	printf("\n");
}

int main() {
	int list[5] = {5, 4, 3, 2, 1};
	bubbleSort(list, 5);
	print(list, 5);
}

File: BubbleSort/bubbleSort.js
var js_test = require('./../Test/JavascriptTest')

function bubble_sort(nums) {
	for (var i = 0; i < nums.length; i++) {
		for (var j = i+1; j < nums.length; j++) {
			if (nums[j] < nums[i]) {
				var temp = nums[i]
				nums[i] = nums[j]
				nums[j] = temp
			}
		}
	}
	return nums
}

console.log(js_test.test(bubble_sort))